### ISSUE TITLE
Functions aren't getting overridden?

### ISSUE THREAD

**Issue by rightaway** [jonschlinkert/assign-deep#6]
I have some function definitions specified in objects and would like them to get overridden when objects are being combined using assign-deep, but it seems to work for all object keys except those referring to function definitions.

```js
a = {
  string: 'a'
  func: a => a,
}
b = {
  string: 'b'
  func: b => b,
}
c = assignDeep({}, a, b)

console.log(c.string) // prints 'b' as expected
console.log(c.func.toString()) // prints 'a => a', not as expected
```

**Comment by jonschlinkert** [comment]
Hmm, I agree that doesn't seem right. I'll take a look, thanks

**Comment by jonschlinkert** [comment]
I think this is the offending code, https://github.com/jonschlinkert/assign-deep/blob/master/index.js#L43. But I can't remember why it's done that way.

**Comment by rightaway** [comment]
Yeah, it only seems to work when `a` doesn't have `func` but doesn't if both `a` and `b` have `func`. Can't think of any reason for it to operate that way. Do you think you would provide a fix for it, is it even a bug?

Btw if I use `Object.assign` it works as expected on the top level of the object (overwriting the function in the result object).

**Comment by jonschlinkert** [comment]
> do you think you would provide a fix for it, is it even a bug?

I can't think of a reason that it should work the way it does, but I haven't had a chance to review the unit tests.  I plan to push up a fix

**Comment by nnuzi** [comment]
I have worked for it and it still WIP. If you like to see: https://github.com/Tommy-White/deeplyAssign
![image](https://user-images.githubusercontent.com/24240963/53064589-547a8380-3503-11e9-9780-77b21e49fee0.png)

**Comment by nnuzi** [comment]
@jonschlinkert
It has been basically completed so far。
